OpenCL_C.txt

```diff
@@ -9571,9 +9571,7 @@ The format is composed of zero or more directives: ordinary characters (not
 specifications, each of which results in fetching zero or more subsequent
 arguments, converting them, if applicable, according to the corresponding
 conversion specifier, and then writing the result to the output stream.
-The format is in the constant address space and must be resolvable at
-compile time, i.e. cannot be dynamically created by the executing program
-itself.
+The format must be a string literal.
```

Comment:
I agree with @SunSerega that this is a bit too strong, at least based on the current Clang behavior, which does not require the format string to be a string literal: https://godbolt.org/z/b33jKTe7K

Is there a C or C++ term we could use instead?

Reply:
I see that Clang allows this, but it isn't happy either:
> warning: format string is not a string literal (potentially insecure)

Doesn't this warning hint that format should be restricted to string literals?

Reply:
The warning is a security-related warning, because in the absence of the constant address space (say, for CPU host code), a malicious actor could modify the content of the format string to do devious things.

I don't think it's relevant to the OpenCL C case, because the string is in the constant address space.  Nevertheless, the warning can be fixed with one more `const`: https://godbolt.org/z/3MYTWWs1P